### PR TITLE
Destroy closed dialogs

### PR DESCRIPTION
**Pull request by jpetrell**

### CODE REVIEW

**Comment by jpetrell** [comment]
No longer WIP. Tested all the affected dialogs keep working.

**Review by blammit — CHANGES_REQUESTED** [review]
Nice, looks great. The only problem is that this doesn't load on wasm, and I suspect it's because `Dialog.qml` is a singleton. There's an issue in wasm builds where QML singletons cannot import `Victron.VenusOS`; this is why we have `Global.qml` with a bunch of global-like properties, which refer to a single instance of various types, that can then be referenced by types that import the `Victron.VenusOS` module.

So I think this is fixable by removing `Dialog.qml` and moving the `open()` function into `pages/DialogLayer.qml`, and then we should be able to call `Global.dialogLayer.open()` from everywhere instead.

File: components/Dialog.qml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+** See LICENSE.txt for license information.
+*/
+pragma Singleton
+
+import QtQuick
+import Victron.VenusOS
+
+QtObject {
+	function open(dialogComponent, properties) {
+		var dialog = dialogComponent.createObject(Global.dialogLayer)
```
[the remark this answers is not in the document]
Comment (jpetrell) [reply components/Dialog.qml]: Yes! I am a bit rusty. :smile: Fixed.

**Comment by jpetrell** [comment]
Fixed the singleton issue.

File: pages/DialogLayer.qml [hunk 2]
```diff
@@ -9,31 +9,13 @@ import Victron.VenusOS
...
 
-	property ModalWarningDialog _rebootDialog
-
-	function showRebootDialog() {
-		if (!_rebootDialog) {
-			_rebootDialog = rebootDialogComponent.createObject(root)
-		}
-		_rebootDialog.open()
+	function open(dialogComponent, properties) {
+		var dialog = dialogComponent.createObject(Global.dialogLayer, properties)
```
Comment (blammit) [inline pages/DialogLayer.qml:13]: I've been preferring `const` and `let` to the more open-scoped `var`.

**Review by blammit — APPROVED** [review]
Minor comment, otherwise LGTM

**Pull request merged**